app/services/group_stage_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class GroupStageService
+  def self.generate_group_stage(groups)
+    return false if groups.length.zero?
+
+    average_group_size = (groups.flatten.length.to_f / groups.length.to_f)
+    if (average_group_size % 1).zero?
+      groups = groups.map { |group| get_group_object_from(group) }
+      group_stage = Stage.new level: -1, groups: groups
+      group_stage
+    else
+      false
+    end
+  end
+
+  def self.get_group_object_from(team_array)
+    matches = generate_all_matches_between team_array
+    group = Group.new matches: matches
+    group
+  end
+
+  def self.generate_all_matches_between(teams)
+    matches = []
+    matchups = teams.combination(2).to_a
```

Comment:
unnecessary var assignment...

Reply:
I need to define `matches` here, as I put stuff into it later while in the loop. As there is no declaration in ruby, I assigned a empty array to it. What would you do instead?

Reply:
Readability